Tests/Swift Tests/BugsnagSwiftPublicAPITests.swift

```diff
@@ -0,0 +1,368 @@
+        config.onCrashHandler = { (writer) in }
+        let crashHandler: (@convention(c)(UnsafePointer<BSG_KSCrashReportWriter>) -> Void)? = { writer in }
+        config.onCrashHandler = crashHandler
+        
+        config.autoDetectErrors = true
+        config.autoTrackSessions = true
+        config.enabledBreadcrumbTypes = .all
+        config.bundleVersion = nil
+        config.bundleVersion = "bundle"
+        config.appType = nil
+        config.appType = "appType"
+        config.maxBreadcrumbs = 999
+        config.persistUser = true
+        
+        let errorTypes =  BugsnagErrorTypes()
+        errorTypes.cppExceptions = true
+        errorTypes.ooms = true
+        errorTypes.machExceptions = true
+        errorTypes.signals = true
+        errorTypes.unhandledExceptions = true
+        errorTypes.unhandledRejections = true
+        config.enabledErrorTypes = errorTypes
+        
+        config.endpoints = BugsnagEndpointConfiguration()
+        config.endpoints = BugsnagEndpointConfiguration(notify: "http://test.com", sessions: "http://test.com")
+        
+        config.setUser("user", withEmail: "email", andName: "name")
+        config.addOnSession(block: sessionBlock)
+        config.removeOnSession(block: sessionBlock)
+        config.addOnSendError(block:onSendErrorBlock)
+        config.removeOnSendError(block: onSendErrorBlock)
+        config.addOnBreadcrumb(block: onBreadcrumbBlock)
+        config.removeOnBreadcrumb(block: onBreadcrumbBlock)
+        
+        let plugin = FakePlugin()
+        config.add(plugin)
+    }
+    
+    // Also test <BugsnagMetadataStore> behaviour
+    func testBugsnagMetadataClass() throws {
+        var md = BugsnagMetadata()
+        md = BugsnagMetadata(dictionary: ["foo" : "bar"])
+        
+        md.addMetadata(["key" : "secret"], section: "mental")
+        md.addMetadata("spock", key: "kirk", section: "enterprise")
+        md.getMetadata(section: "mental")
+        md.getMetadata(section: "mental", key: "key")
+        md.clearMetadata(section: "enterprise")
+        md.clearMetadata(section: "enterprise", key: "key")
+    }
+    
+    func testBugsnagEventClass() throws {
+        let event = BugsnagEvent()
+        
+        event.context = nil
+        event.context = "ctx"
+        event.errors = []
+        event.errors = [BugsnagError()]
+        event.groupingHash = nil
+        event.groupingHash = "1234"
+        event.breadcrumbs = []
+        event.breadcrumbs = [BugsnagBreadcrumb()]
+        event.apiKey = apiKey
+        _ = event.device
+        _ = event.app
+        _ = event.unhandled
+        event.threads = []
+        event.threads = [BugsnagThread()]
+        event.originalError = nil
+        event.originalError = 123
+        event.originalError = BugsnagError()
+        _ = event.user
+        event.setUser("user", withEmail: "email", andName: "name")
+    }
+    
+    func testBugsnagAppClass() throws {
+        let app = BugsnagApp()
```

Comment:
`BugsnagAppWithState` and `BugsnagDeviceWithState` should be checked instead as they include additional stateful fields